### PR TITLE
Chap 4 Word2Vec Chinese Version 

### PR DESCRIPTION
**Pull request by daming-lu**

### CODE REVIEW

File: 04.word2vec/README.cn.md [hunk 1]
```diff
@@ -196,236 +197,236 @@ dream that one day <e>
...
-
-- 将文本隐层特征，再经过一个全连接，映射成一个$|V|$维向量，同时通过softmax归一化得到这`|V|`个词的生成概率。
+- 我们来定义我们的 N-gram 神经网络结构。这个结构在训练和预测中都会使用到。
 
 ```python
-predictword = paddle.layer.fc(input=hidden1,
-                              size=dict_size,
-                              bias_attr=paddle.attr.Param(learning_rate=2),
-                              act=paddle.activation.Softmax())
+def inference_program(is_sparse):
```
Comment (jetfuel) [inline 04.word2vec/README.cn.md:228]: The `is_sparse` is used every where, but not explained. It will be nice if we can briefly explain it.

File: 04.word2vec/README.cn.md [hunk 2]
```diff
@@ -196,236 +197,236 @@ dream that one day <e>
...
+    # thirdw, fourthw], which is not correct.
+    predict_word = inference_program(is_sparse)
+    next_word = fluid.layers.data(name='nextw', shape=[1], dtype='int64')
+    cost = fluid.layers.cross_entropy(input=predict_word, label=next_word)
+    avg_cost = fluid.layers.mean(cost)
+    return avg_cost
 ```
 
-然后，指定训练相关的参数：
+- 现在我可以开始训练啦。如今的版本较之以前就简单了许多。我们有现成的训练和测试集：`paddle.dataset.imikolov.train()`和`paddle.dataset.imikolov.test()`。两者都会返回一个读取器。在PaddlePaddle中，读取器是一个Python的函数，每次调用，会读取下一条数据。它是一个Python的generator。
```
Comment (jetfuel) [inline 04.word2vec/README.cn.md]: 我 -> 我们

File: 04.word2vec/README.cn.md [hunk 3]
```diff
@@ -196,236 +197,236 @@ dream that one day <e>
...
+    train_reader = paddle.batch(
+        paddle.dataset.imikolov.train(word_dict, N), BATCH_SIZE)
+    test_reader = paddle.batch(
+        paddle.dataset.imikolov.test(word_dict, N), BATCH_SIZE)
+
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+
+    def event_handler(event):
+        if isinstance(event, fluid.EndStepEvent):
+            outs = trainer.test(
```
Comment (jetfuel) [inline 04.word2vec/README.cn.md]: We can put all the code under the `if event.step % 10 == 0:` block.

File: 04.word2vec/README.cn.md [hunk 4]
```diff
@@ -196,236 +197,236 @@ dream that one day <e>
...
+
+            if math.isnan(avg_cost):
+                sys.exit("got NaN loss, training failed.")
+
+    trainer = fluid.Trainer(
+        train_func=train_program,
+        # Note here we need to choose more sophisticated optimizer
+        # such as AdaGrad with a decay rate. The normal SGD converges
+        # very slowly.
+        # optimizer=fluid.optimizer.SGD(learning_rate=0.001),
```
Comment (jetfuel) [inline 04.word2vec/README.cn.md]: We need to update this comment. 
or move the whole thing to `optimizer_func`

**Review by jetfuel — APPROVED** [review]
Otherwise LGTM
